src/Tools/AI Test Toolkit/src/TestSuite/AITALTestSuiteMgt.Codeunit.al

```diff
@@ -0,0 +1,202 @@
+    begin
+        TestMethodLine.SetRange("Test Suite", ALTestSuite.Name);
+        TestMethodLine.SetRange("Line Type", TestMethodLine."Line Type"::Codeunit);
+        TestMethodLine.ReadIsolation := TestMethodLine.ReadIsolation::ReadUncommitted;
+
+        if not TestMethodLine.FindSet() then
+            exit;
+
+        repeat
+            FunctionTestMethodLine.SetRange("Test Suite", ALTestSuite.Name);
+            FunctionTestMethodLine.SetRange("Test Codeunit", TestMethodLine."Test Codeunit");
+            FunctionTestMethodLine.SetRange("Line Type", FunctionTestMethodLine."Line Type"::Function);
+            FunctionTestMethodLine.ReadIsolation := FunctionTestMethodLine.ReadIsolation::ReadUncommitted;
+            if FunctionTestMethodLine.IsEmpty() then
+                TestMethodLine.Delete();
+        until TestMethodLine.Next() = 0;
+    end;
+
+    internal procedure GetOrCreateALTestSuite(var AITTestMethodLine: Record "AIT Test Method Line"): Record "AL Test Suite"
+    var
+        AITTestSuite: Record "AIT Test Suite";
+        ALTestSuite: Record "AL Test Suite";
+    begin
+        if AITTestMethodLine."AL Test Suite" <> '' then begin
+            ALTestSuite.SetFilter(Name, AITTestMethodLine."AL Test Suite");
+            if ALTestSuite.FindFirst() then
+                exit(ALTestSuite);
+        end;
+
+        if AITTestMethodLine."AL Test Suite" = '' then begin
+            AITTestMethodLine."AL Test Suite" := this.GetUniqueAITTestSuiteCode();
+            AITTestMethodLine.Modify();
+        end;
+
+        ALTestSuite.Name := AITTestMethodLine."AL Test Suite";
+        ALTestSuite.Description := CopyStr(AITTestMethodLine.Description, 1, MaxStrLen(ALTestSuite.Description));
+        AITTestSuite.ReadIsolation := IsolationLevel::ReadUncommitted;
+        if AITTestSuite.Get(AITTestMethodLine."Test Suite Code") then
+            ALTestSuite."Test Runner Id" := AITTestSuite."Test Runner Id";
+
+        ALTestSuite.Insert(true);
+        exit(ALTestSuite);
+    end;
+
+    local procedure GetUniqueAITTestSuiteCode(): Code[10]
+    var
+        ALTestSuite: Record "AL Test Suite";
+    begin
+        ALTestSuite.SetFilter(Name, this.AITTestSuitePrefixLbl + '*');
+        if not ALTestSuite.FindLast() then
+            exit(this.AITTestSuitePrefixLbl + '000001');
+
+        exit(IncStr(ALTestSuite.Name))
+    end;
+
+    internal procedure DownloadTestOutputFromLogToFile(var AITLogEntry: Record "AIT Log Entry")
+    var
+        TempBlob: Codeunit "Temp Blob";
+        TestOutput: Text;
+        FileNameTxt: Text;
+        JsonTextBuilder: TextBuilder;
+        JsonOutStream: OutStream;
+        JsonInStream: InStream;
+        NoTestOutputFoundErr: Label 'No Test Output found in the logs';
+    begin
+        AITLogEntry.SetLoadFields("Test Suite Code", "Output Data");
+        if AITLogEntry.FindSet(false) then begin
+            FileNameTxt := Format(AITLogEntry."Test Suite Code") + '_' + 'test_output' + '.jsonl';
```

Comment:
Use StrSubstNo for string concatenation.

Also, is the extension ".jsonl" correct?

Reply:
.jsonl is correct, it is Json New Line